modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/BlackFxSingleBarrierOptionProductPricer.java

```diff
@@ -0,0 +1,450 @@
+import com.opengamma.strata.pricer.impl.option.BlackOneTouchCashPriceFormulaRepository;
+import com.opengamma.strata.pricer.rate.RatesProvider;
+import com.opengamma.strata.product.fx.ResolvedFxSingle;
+import com.opengamma.strata.product.fx.ResolvedFxSingleBarrierOption;
+import com.opengamma.strata.product.fx.ResolvedFxVanillaOption;
+import com.opengamma.strata.product.fx.SimpleConstantContinuousBarrier;
+
+/**
+ * Pricer for FX barrier option products in Black-Scholes world.
+ * <p>
+ * This function provides the ability to price an {@link ResolvedFxSingleBarrierOption}.
+ * <p>
+ * All of the computation is be based on the counter currency of the underlying FX transaction. 
+ * For example, price, PV and risk measures of the product will be expressed in USD for an option on EUR/USD.
+ */
+public class BlackFxSingleBarrierOptionProductPricer {
+
+  /**
+   * Default implementation.
+   */
+  public static final BlackFxSingleBarrierOptionProductPricer DEFAULT = new BlackFxSingleBarrierOptionProductPricer();
+
+  /**
+   * Pricer for barrier option without rebate.
+   */
+  private static final BlackBarrierPriceFormulaRepository BARRIER_PRICER = new BlackBarrierPriceFormulaRepository();
+  /**
+   * Pricer for rebate.
+   */
+  private static final BlackOneTouchAssetPriceFormulaRepository ASSET_REBATE_PRICER = new BlackOneTouchAssetPriceFormulaRepository();
+  /**
+   * Pricer for rebate.
+   */
+  private static final BlackOneTouchCashPriceFormulaRepository CASH_REBATE_PRICER = new BlackOneTouchCashPriceFormulaRepository();
+
+  /**
+   * Creates an instance.
+   */
+  public BlackFxSingleBarrierOptionProductPricer() {
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value of the FX barrier option product.
+   * <p>
+   * The present value of the product is the value on the valuation date. 
+   * It is expressed in the counter currency.
+   * 
+   * @param option  the option product
+   * @param ratesProvider  the rates provider
+   * @param volatilityProvider  the Black volatility provider
+   * @return the present value of the product
+   */
+  public CurrencyAmount presentValue(
+      ResolvedFxSingleBarrierOption option,
+      RatesProvider ratesProvider,
+      BlackVolatilityFxProvider volatilityProvider) {
+
+    double price = price(option, ratesProvider, volatilityProvider);
+    ResolvedFxVanillaOption underlyingOption = option.getUnderlyingOption();
+    return CurrencyAmount.of(underlyingOption.getCounterCurrency(), signedNotional(underlyingOption) * price);
+  }
+
+  /**
+   * Calculates the price of the FX barrier option product.
+   * <p>
+   * The price of the product is the value on the valuation date for one unit of the base currency 
+   * and is expressed in the counter currency. The price does not take into account the long/short flag. 
+   * See {@link #presentValue} for scaling and currency.
+   * 
```

Comment:
Add explanation on how the volatility is selected. As there is a strike and a barrier, it is not obvious which number is used.